packages/plugins/RedPacket/src/SiteAdaptor/RedPacketERC721Form.tsx

```diff
@@ -391,62 +403,71 @@ export function RedPacketERC721Form(props: RedPacketERC721FormProps) {
                         chainId={chainId}
                     />
                 </Box>
-                {collection && balance ? (
-                    <Box className={classes.selectWrapper}>
-                        <div
-                            className={cx(
-                                classes.optionLeft,
-                                classes.option,
-                                balance === 0 ? classes.disabledSelector : null,
-                            )}
-                            onClick={() => {
-                                setSelectOption(NFTSelectOption.All)
-                                setExistTokenDetailedList(tokenDetailedOwnerList.slice(0, maxSelectShares))
-                                setAllTokenDetailedList(tokenDetailedOwnerList.slice(0, maxSelectShares))
-                            }}>
+                {collection && balance && done ? (
```

Comment:
`collection && balance && done`

It might take very long time if user has lots of NFTs.

You don't need to fetch all assets before setting all.

`All` is just a flag. If the flag is on, you can pass all the NFTs, and you can fetch all NFTs before submit.

Don't need to maintain a big list for selected items, just maintain the selectedIds.

```tsx
const [isAll, setIsAll] = useState(false)
const [selectedIds, setSelectedIds] = useState<string[]>([])
const { dat: allNFTs } = useQueryAllNFTS()
const allIds = useMemo(() => allNFTs.map(x => x.id), [allNFTs])
const isAllSelected = useMemo(() => {
  return isAll || isEqual(selectedIds, allIds,
  }, [selectedIds, allIds])


return <div>
{isAllSelected ? 'all' : 'partcial'}
  <list>
  {allNFTs.map(nft => (
    <item>{isAll || selectedIds.includes(nft.id) ? 'selected' : 'unselected'}</item>
  )}
<list>
</div>

````